elements/lisk-chain/src/state_store/state_store.ts

```diff
@@ -222,14 +224,21 @@ export class StateStore {
 	// createSnapshot follows the same interface as stateDB. However, it does not support multi snapshot.
 	public createSnapshot(): number {
 		this._snapshot = this._cache.copy();
-		return 0;
+		this._latestSnapshotId += 1;
```

Comment:
if the system creates snapshot for very long time, the id will overflow the limit?

Reply:
Is there a limit that we should take into account?

Reply:
The limit for the number type is `Number.MAX_VALUE` => `1.7976931348623157e+308`. However, Numbers larger than MAX_VALUE are represented as Infinity.

Reply:
assuming the node will run forever, we need to reset the number within the range

Reply:
What range would that be?

Reply:
the range is MAX_VALUE. ie) the number shouldnt exceed the max value

Reply:
Just to clarify your statement (...we need to reset the number within the range), If it exceeds `Number.MAX_VALUE`, reset to 0 or any number within the range?

Reply:
in this case, you would want to use `MAX_SAFE_INTEGER` since it's integer. I think it make sense to reset to `0` if you want to use it as a counter